frontend/src/components/modals/AddCustomerModal.tsx

```diff
@@ -0,0 +1,250 @@
+import { randomColor, getCheckedIds } from '../../utils/CustomerUtils';
+import css from './AddCustomerModal.module.scss';
+
+const classes = classNames.bind(css);
+
+const steps = [
+  { step: 1, description: 'Client info' },
+  { step: 2, description: 'Select representative' },
+  { step: 3, description: 'Finish' },
+];
+
+export const AddCustomerModal: React.FC<ModalProps> = ({ closeModal }) => {
+  const dispatch = useDispatch<StoreDispatchType>();
+  const customers = useSelector<RootState, Customer[] | undefined>(
+    allCustomersSelector,
+    shallowEqual,
+  );
+  const roadmapUsers = useSelector<RootState, RoadmapUser[] | undefined>(
+    roadmapUsersSelector,
+    shallowEqual,
+  );
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [colorType, setColorType] = useState('generate');
+  const [formValues, setFormValues] = useState({
+    name: '',
+    email: '',
+    color: randomColor(customers),
+  });
+  const [representatives, setRepresentatives] = useState<CheckableUser[]>([]);
+  const [step, setStep] = useState(1);
+  const [previousStep, setPreviousStep] = useState(0);
+
+  useEffect(() => {
+    if (!roadmapUsers) return;
+    setRepresentatives(
+      roadmapUsers
+        ?.filter(
+          (user) =>
+            user.type === RoleType.Admin || user.type === RoleType.Business,
+        )
+        .map((obj) => ({ ...obj, checked: false })),
+    );
+  }, [roadmapUsers]);
+
+  useEffect(() => {
+    if (!roadmapUsers) dispatch(roadmapsActions.getRoadmapUsers());
+  }, [dispatch, roadmapUsers]);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    switch (step) {
+      case 0:
+        closeModal();
+        break;
+      case 1:
+        setStep(2);
+        break;
+      case 2:
+        if (!getCheckedIds(representatives).length) {
+          setErrorMessage('At least one representative needs to be selected');
+          break;
+        }
+        setErrorMessage('');
+        setStep(3);
+        break;
+      case 3: {
+        setIsLoading(true);
+
+        const res = await dispatch(
+          roadmapsActions.addCustomer({
+            name: formValues.name,
+            email: formValues.email,
+            color:
+              colorType === 'pick' ? formValues.color : randomColor(customers),
+            representatives: getCheckedIds(representatives),
+          }),
+        );
+        setIsLoading(false);
+        if (roadmapsActions.addCustomer.rejected.match(res)) {
+          if (res.payload?.message) setErrorMessage(res.payload.message);
+          return;
+        }
+        setStep(step + 1);
+        break;
+      }
+      default:
+        closeModal();
+    }
+  };
+
+  const handleCancel = () => {
+    if (step === 0) {
+      setStep(previousStep);
+      return;
+    }
+    setPreviousStep(step);
+    setStep(0);
+  };
```

Comment:
Cancel sulkee nyt koko modaalin, menee mielestäni vähän ristiin lopussa olevan tekstin kanssa "or go back to previous steps". Saisi kuvan, että pystyisi liikkumaan taaksepäin muokkaamaan/korjaamaan tietoja

Reply:
Kai cancel näyttää kuitenkin kuvan varoituksen?
![image](https://user-images.githubusercontent.com/80758782/123757123-53717200-d8c6-11eb-867e-49c988c0baca.png)
Vihreistä stepeistä klikkaamalla pääsee liikkumaan taaksepäin. Mielestäni modaali toimii kuten figmassa. Toisaalta, jos logiikka on epäintuitiivinen ilman selitystä, niin voidaan miettiä miten sitä parannettaisiin
![image](https://user-images.githubusercontent.com/80758782/123757193-62f0bb00-d8c6-11eb-937b-7e80efff92cc.png)

Reply:
Näyttää 👍